### PR TITLE
feat(Catalog Management): re-gen service after recent API changes

### PR DESCRIPTION
**Pull request by benbuchanan**
## PR summary
This PR is to update our node SDK after many API changes since the last update.

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Current vs new behavior  
Includes many new features and capabilities to the catalog management API

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by benbuchanan** [comment]
Example tests:
```
Test Suites: 1 passed, 1 total
Tests:       30 passed, 30 total
Snapshots:   0 total
Time:        16.366 s, estimated 19 s
Ran all test suites matching /examples\/catalog-management.v1.test.js/i.
```

Integration tests:
```
Test Suites: 18 skipped, 1 passed, 1 of 19 total
Tests:       373 skipped, 33 passed, 406 total
Snapshots:   0 total
Time:        18.803 s, estimated 20 s
Ran all test suites matching /test\/integration\//i.
```

File: examples/catalog-management.v1.test.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
  * @jest-environment node
```
Comment (padamstx) [inline examples/catalog-management.v1.test.js:2]: This file is missing example code blocks for deleteShareApprovalList() and getOfferingSourceArchive()

File: test/integration/catalog-management.v1.test.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
-/* eslint-disable no-console */
```
Comment (padamstx) [inline test/integration/catalog-management.v1.test.js:1]: This file is missing tests for deleteShareApprovalList() and getOfferingSourceArchive().

**Review by padamstx — COMMENTED** [review]
Overall it looks fine.  I just noticed that the examples and integration test files are missing tests for a couple of operations.  We really should have examples and integration tests for all operations.

**Review by padamstx — APPROVED** [review]
LGTM

**Pull request merged**